src/Symfony/Component/Form/Extension/Core/DataTransformer/DateTimeToLocalizedStringTransformer.php

```diff
@@ -130,14 +130,14 @@ public function reverseTransform($value)
         try {
             if ($dateOnly) {
                 // we only care about year-month-date, which has been delivered as a timestamp pointing to UTC midnight
-                return new \DateTime(gmdate('Y-m-d', $timestamp), new \DateTimeZone($this->inputTimezone));
+                $dateTime = new \DateTime(gmdate('Y-m-d', $timestamp), new \DateTimeZone($this->outputTimezone));
+            } else {
+                // read timestamp into DateTime object - the formatter delivers a timestamp
+                $dateTime = new \DateTime(sprintf('@%s', $timestamp));
+                // set timezone separately, as it would be ignored if set via the constructor,
+                // see http://php.net/manual/en/datetime.construct.php
+                $dateTime->setTimezone(new \DateTimeZone($this->outputTimezone));
```

Comment:
I see why I'm confused: since we pass `(gmdate('Y-m-d', $timestamp)` as first argument in the construct line 133 I'm assuming that there will never be a timezone there and you don't. (am I right?)

I'll change accordingly.